src/lib/rpmostree-package.c

```diff
@@ -258,29 +214,21 @@ _rpm_ostree_package_list_for_commit (OstreeRepo   *repo,
   if (!ostree_repo_load_variant (repo, OSTREE_OBJECT_TYPE_COMMIT, checksum, &commit, error))
     return FALSE;
 
+  /* We used to have a fallback here to checking out the rpmdb from the commit,
+   * but that currently drags in our internal Rust code which bloats this
+   * shared library and causes other problems since the main executable
+   * wants to link to this shared library too.  So for now if we don't
+   * find the pkglist in the commit metadata, just return as if it's
+   * empty.
+   *
+   * TODO: Rework this to run via the special shlib-ipc that is used
+   * in rpmostree.c.
+   */
   g_autoptr(GVariant) pkglist_v = get_commit_rpmdb_pkglist (commit);
   if (!pkglist_v)
     {
-      /* file-based rpmdb fallback; let fail if rpmdb not available */
-      g_autoptr(GError) local_error = NULL;
-      g_autoptr(RpmOstreeRefSack) rsack =
-        rpmostree_get_refsack_for_commit (repo, rev, cancellable, &local_error);
-      if (!rsack)
-        {
-          if (allow_noent &&
-              g_error_matches (local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
-            {
-              *out_pkglist = NULL;
-              return TRUE; /* NB: early return */
-            }
-
-          g_propagate_error (error, g_steal_pointer (&local_error));
-          return FALSE;
-        }
-
-      /* Note early return */
-      *out_pkglist = query_all_packages_in_sack (rsack);
-      return TRUE;
```

Comment:
I think this is *probably* OK to drop, but I wouldn't necessarily qualify it as a purely "backcompat" thing. Currently the pkglist lives in the commit metadata, and sadly it's very easy to lose this when shuffling OSTree commits around across different branches, repos, etc...

Obviously, workflows can adapt. But it wouldn't surprise me if some people are unknowingly relying on this right now when they do `rpm-ostree db list/diff` on their OSTree repo. Overall, I think the rpmdb fallback is still worth keeping? Another way I look at this is that the pkglist metadata is an (important) optimization to avoid querying the source of truth (the rpmdb).

Reply:
With the socket backend, can we generate the pkglist from the rpmdb (using `rpmostree_create_rpmdb_pkglist_variant`) and pass that back to the library here for the fallback?

Reply:
Right, https://github.com/coreos/coreos-assembler/issues/1964 is in fact one where this would probably trigger.

Reply:
OK I'll give it a shot.  At least it'll mean the gvariant-over-memfd logic will actually be worth it instead of massive overkill :smile: